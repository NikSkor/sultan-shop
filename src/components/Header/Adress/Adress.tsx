import React, { FC } from 'react';
import style from './Adress.module.scss';

interface AdressProps {
  title: string,
  description: string,
  children?: React.ReactNode
  titleColor: string,
  descrColor: string,
}

const Adress: FC<AdressProps> = ({title, description, titleColor, descrColor, children}) => {
  return (
    <div className={style.block}>
      {children}
      <div className={style.info}>
        <div className={style.title} style={{color: titleColor}}>{title}</div>
        <div className={style.description} style={{color: descrColor}}>{description}</div>
      </div>
    </div>
  )
}

export default Adress;