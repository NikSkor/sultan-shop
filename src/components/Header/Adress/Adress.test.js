import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Adress from '../Adress/Adress.tsx';

afterEach(cleanup);

  describe('Adress.tsx', () => {
    it('component renders props correctly, truthy values', () => {
      const tree = renderer
        .create(
          <Adress
            title='г. Кокчетав, ул. Ж. Ташенова 129Б'
            description='(Рынок Восточный)'
            titleColor='#111111'
            descrColor='#3F4E65'
          />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('component renders props correctly, falsy values', () => {
      const tree = renderer.create(<Adress />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

