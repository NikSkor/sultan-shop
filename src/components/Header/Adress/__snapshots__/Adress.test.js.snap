// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adress.tsx component renders props correctly, falsy values 1`] = `
<div
  className="block"
>
  <div
    className="info"
  >
    <div
      className="title"
      style={
        Object {
          "color": undefined,
        }
      }
    />
    <div
      className="description"
      style={
        Object {
          "color": undefined,
        }
      }
    />
  </div>
</div>
`;

exports[`Adress.tsx component renders props correctly, truthy values 1`] = `
<div
  className="block"
>
  <div
    className="info"
  >
    <div
      className="title"
      style={
        Object {
          "color": "#111111",
        }
      }
    >
      г. Кокчетав, ул. Ж. Ташенова 129Б
    </div>
    <div
      className="description"
      style={
        Object {
          "color": "#3F4E65",
        }
      }
    >
      (Рынок Восточный)
    </div>
  </div>
</div>
`;
